rasa/src/Rasa/Internal/Directive.hs

```diff
@@ -127,7 +127,10 @@ replaceRange r txt = range r .= txt
 
 -- | Inserts text into the buffer at the given Coord.
 insertAt :: Coord -> Y.YiString -> BufAction ()
-insertAt c txt = range (Range c c) .= txt
+insertAt c txt = do
+  let r = Range c c
+  liftAction $ dispatchEvent (BufTextChanged r)
```

Comment:
We'll need to add this dispatch to `overRange` and `replaceRange` as well since those both edit the buffer's text.

Or; better yet we can move the dispatch to just be in `overRange` and implement the others in terms of it:

```haskell
insertAt c txt =   replaceRange r txt
  where r = Range c c

replaceRange r txt = overRange r (const txt)
```

Make sense? 😃 👍